integrated_channels/integrated_channel/exporters/content_metadata.py

```diff
@@ -71,28 +75,44 @@ def __init__(self, user, enterprise_configuration):
 
     def export(self, **kwargs):
         """
-        Return the exported and transformed content metadata as a dictionary.
+        Export transformed content metadata if there has been an update to the consumer's catalogs
         """
+        enterprise_customer_catalogs = self.enterprise_configuration.customer_catalogs_to_transmit or \
+            self.enterprise_customer.enterprise_customer_catalogs.all()
+
         content_metadata_export = {}
-        content_metadata_items = self.enterprise_catalog_api.get_content_metadata(
-            self.enterprise_customer,
-            enterprise_catalogs=self.enterprise_configuration.customer_catalogs_to_transmit
-        )
-        LOGGER.info(
-            'Getting metadata for Enterprise [%s], Catalogs [%s] from Enterprise Catalog Service. Results: [%s]',
-            self.enterprise_customer.name,
-            self.enterprise_configuration.customer_catalogs_to_transmit,
-            json.dumps(content_metadata_items)
-        )
-        for item in content_metadata_items:
-            transformed = self._transform_item(item)
-            LOGGER.debug(
-                'Exporting content metadata item with plugin configuration [%s]: [%s]',
-                self.enterprise_configuration,
-                json.dumps(transformed, indent=4),
-            )
-            content_metadata_item_export = ContentMetadataItemExport(item, transformed)
-            content_metadata_export[content_metadata_item_export.content_id] = content_metadata_item_export
+        force_retrieve_all_catalogs = kwargs.get('force_retrieve_all_catalogs')
+        for enterprise_customer_catalog in enterprise_customer_catalogs:
+            if not force_retrieve_all_catalogs:
+                # Retrieve the time at which the catalog and it's content were last updated
+                content_last_modified, catalog_modified = self._get_enterprise_catalog_last_changed(
+                    enterprise_customer_catalog
+                )
+                # Retrieve whatever modification happened more recently
+                catalog_last_modified = max(
+                    content_last_modified,
+                    catalog_modified
+                )
+                last_successful_transmission = self._get_most_recent_catalog_update_time()
+
+            # If the last successful transmission was since the last time the catalog was updated, there's no need for
+            # an update
+            if force_retrieve_all_catalogs or not last_successful_transmission or \
```

Comment:
I know this will work b/c force_retrieve_all_catalogs means you won't eval the rest of the if statement, but it's really hard to figure out that if statement when half the time the variables in it will be undefined. Can you abstract out it:

```
need_catalog_update = False
if not force_retrieve_all_catalogs:
  do_stuff;
  need_catalog_update = (not last_successful_transmission) or (last_successful_transmission < catalog_last_modified)
  
 if not force_retrieve_all_catalogs or need_catalog_update: 
   do_more_stuff()
```